lib/sidekiq/redis_connection.rb

```diff
@@ -7,10 +7,7 @@ class RedisConnection
     class << self
 
       def create(options={})
-        url = options[:url] || determine_redis_provider
-        if url
-          options[:url] = url
-        end
+        options[:url] ||= determine_redis_provider
```

Comment:
@mperham as I can see, `url` options uses only in [`log_info`](https://github.com/mperham/sidekiq/blob/master/lib/sidekiq/redis_connection.rb#L60-L63) method:
``` ruby
if scrubbed_options[:url] && (uri = URI.parse(scrubbed_options[:url])) && uri.password
  uri.password = redacted
  scrubbed_options[:url] = uri.to_s
end
```
And I think that's not important to set `:url => nil` in option hash or not